packages/app/src/flame/valibot/flameSchema.ts

```diff
@@ -0,0 +1,96 @@
+import * as v from 'valibot'
+import { TransformVariationSchema } from './variationSchema'
+import type { DrawMode } from '../drawMode'
+
+// default values and schema fallbacks
+const latestValibotVersion = '1.0'
+const flameDescriptorVersionDefault = latestValibotVersion
+export const backgroundColorDefault: [number, number, number] = [0, 0, 0]
+export const backgroundColorDefaultWhite: [number, number, number] = [1, 1, 1]
+const cameraDefault: { zoom: number; position: [number, number] } = {
+  zoom: 1,
+  position: [0, 0],
+}
+const renderSettingsDefult = {
+  exposure: 0.25,
+  skipIters: 20,
+  drawMode: 'light' as DrawMode,
+  backgroundColor: backgroundColorDefault,
+  camera: cameraDefault,
+}
+const metadataDefault = {
+  version: flameDescriptorVersionDefault,
+  author: 'unknown',
+}
+
+export const TransformIdSchema = v.pipe(v.string(), v.brand('TransformId'))
+export const VariationIdSchema = v.pipe(v.string(), v.brand('VariationId'))
+export const VariationTypeSchema = v.pipe(v.string(), v.brand('VariationType'))
+export const DrawModeSchema = v.union([v.literal('light'), v.literal('paint')])
+
+export const AffineParamsSchema = v.object({
+  a: v.number(),
+  b: v.number(),
+  c: v.number(),
+  d: v.number(),
+  e: v.number(),
+  f: v.number(),
+})
+
+const VariationRecordSchema = v.record(
+  VariationIdSchema,
+  TransformVariationSchema,
+)
+export const TransformFunctionSchema = v.object({
+  probability: v.number(),
+  preAffine: AffineParamsSchema,
+  postAffine: AffineParamsSchema,
+  color: v.object({
+    x: v.number(),
+    y: v.number(),
+  }),
+  variations: VariationRecordSchema,
+})
+
+const CameraObjSchema = v.object({
+  zoom: v.optional(v.number(), cameraDefault.zoom),
+  position: v.optional(
+    v.tuple([v.number(), v.number()]),
+    cameraDefault.position,
+  ),
+})
+
+const RenderSettingsSchema = v.optional(
+  v.strictObject({
+    exposure: v.pipe(
+      v.number(),
+      v.minValue(-4, 'The exposure should be in range [-4, 4]'),
```

Comment:
Some extra validation to discuss, we could add a lot of things to ensure proper sane values, this is just an example how this can be done

Reply:
I wouldn't check this here. Nothing bad will happen if the value is outside of this range. But for other things it might make sense, especially performance-sensitive things like `skipIters`

Reply:
Yeah, well skip iters could be one to validate a bit more, even though in theory it could be quite a big number, I'll play a bit with range and set some sensible defaults.